demo/run-osm-multicluster-demo.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+
+# Automated Multicluster Demo - Work in Progress
+# This script will deploy two kind clusters to the local machine.
+# Each cluster will have Open Service Mesh and the various Bookstore containers deployed to them.
+# Following Multicluster feature support - TBD on how we'll demonstrate multicluster access.
+#   Perhaps having the bookthief from one cluster access the bookstore in the other cluster.
+
+set -aueo pipefail
+
+if [ ! -f .env ]; then
+    echo -e "\nThere is no .env file in the root of this repository."
+    echo -e "Copy the values from .env.example into .env."
+    echo -e "Modify the values in .env to match your setup.\n"
+    echo -e "    cat .env.example > .env\n\n"
+    exit 1
+fi
+
+TIMEOUT="600s"
+
+# shellcheck disable=SC1091
+source .env
+
+# Set meaningful defaults for env vars we expect from .env
+CI="${CI:-false}"  # This is set to true by Github Actions
+MESH_NAME="${MESH_NAME:-osm}"
+K8S_NAMESPACE="${K8S_NAMESPACE:-osm-system}"
+BOOKBUYER_NAMESPACE="${BOOKBUYER_NAMESPACE:-bookbuyer}"
+BOOKSTORE_NAMESPACE="${BOOKSTORE_NAMESPACE:-bookstore}"
+BOOKTHIEF_NAMESPACE="${BOOKTHIEF_NAMESPACE:-bookthief}"
+BOOKWAREHOUSE_NAMESPACE="${BOOKWAREHOUSE_NAMESPACE:-bookwarehouse}"
+CERT_MANAGER="${CERT_MANAGER:-tresor}"
+CTR_REGISTRY="${CTR_REGISTRY:-localhost:5000}"
+CTR_REGISTRY_CREDS_NAME="${CTR_REGISTRY_CREDS_NAME:-acr-creds}"
+DEPLOY_TRAFFIC_SPLIT="${DEPLOY_TRAFFIC_SPLIT:-true}"
+CTR_TAG="${CTR_TAG:-$(git rev-parse HEAD)}"
+IMAGE_PULL_POLICY="${IMAGE_PULL_POLICY:-Always}"
+ENABLE_DEBUG_SERVER="${ENABLE_DEBUG_SERVER:-true}"
+ENABLE_EGRESS="${ENABLE_EGRESS:-false}"
+DEPLOY_GRAFANA="${DEPLOY_GRAFANA:-false}"
+DEPLOY_JAEGER="${DEPLOY_JAEGER:-false}"
+ENABLE_FLUENTBIT="${ENABLE_FLUENTBIT:-false}"
+DEPLOY_PROMETHEUS="${DEPLOY_PROMETHEUS:-false}"
+DEPLOY_WITH_SAME_SA="${DEPLOY_WITH_SAME_SA:-false}"
+ENVOY_LOG_LEVEL="${ENVOY_LOG_LEVEL:-debug}"
+DEPLOY_ON_OPENSHIFT="${DEPLOY_ON_OPENSHIFT:-false}"
+
+# For any additional installation arguments. Used heavily in CI.
+optionalInstallArgs=$*
+
+exit_error() {
+    error="$1"
+    echo "$error"
+    exit 1
+}
+
+# Check if Docker daemon is running
+docker info > /dev/null || { echo "Docker daemon is not running"; exit 1; }
+
+# Build OSM binaries
+make build-osm
+
+for CLUSTER in "osm-alpha" "osm-beta"; do
+    KIND_CLUSTER_NAME=$CLUSTER
+    # shellcheck disable=SC2034
+    BOOKSTORE_CLUSTER_ID=$CLUSTER # this is used when deploying bookstore app
+    echo "Creating cluster named: $KIND_CLUSTER_NAME"
+    ./scripts/kind-with-registry.sh
```

Comment:
Lets not force the creation of a cluster on kind. Can we assume the person running this script has two cluster already ? This can be a pre-req just like the run-osm-demo.sh script, where one can use a cluster on any env

Reply:
Removed from this script. In the demo using Kind cluster, we can run this line manually.